apps/extension/src/ui/atoms/tokenRates.ts

```diff
@@ -38,3 +41,48 @@ export const tokenRatesQuery = selectorFamily({
       return tokenId ? tokenRates[tokenId] : undefined
     },
 })
+
+const _selectableCurrenciesState = atom<ReadonlySet<TokenRateCurrency>>({
+  key: "_selectableCurrency",
+  default: new Set(["usd", "dot", "eth"]),
+  effects: [
+    storageEffect(Browser.storage.local, {
+      key: "settings",
+      subKey: "selectableCurrencies",
+      isSet: true,
+    }),
+  ],
+})
+
+export const selectableCurrenciesState = selector<readonly TokenRateCurrency[]>({
```

Comment:
Here we can't just use the hook for plainly getting and setting values though, cause as you can see these atoms have custom logic for get & set value

Reply:
But I have change the atoms to use the existing settings store instead

Reply:
If we need additional logic on top of the `useSetting` hook, it would be better to create a `useCurrencySetting` hook to expose getting and setting that particular value as we have done with `tokenRatesState`. Still not 100% sure this needs to be in its own atom, also - can't we just put this logic into an ordinary hook which calls `useSetting` and exposes that getting and setting logic? Anyway, if it really does need an atom, let's make a new file for that rather than adding it to `tokenRates`, which is related but slightly different data.

Reply:
Similar to stuff like redux & redux saga, one advantage of using state lib like Recoil is the ability to completely extract business logic from the view/render layer. So separation of concern + in the case of recoil, automatic memoization of selector, so computation will only happen once rather than on every re-render.

Reply:
How's another file named `currency.ts` sound?